File: SurveyBuilder/SurveyBuilder.API/Models/Survey.cs
namespace SurveyBuilder.API.Models;

public class Survey
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsPublished { get; set; } = false;
    public string ShareableCode { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public string UserId { get; set; } = string.Empty;
    public User User { get; set; } = null!;

    public ICollection<Question> Questions { get; set; } = [];
    public ICollection<Response> Responses { get; set; } = [];
}

File: SurveyBuilder/survey-builder-client/src/types/survey.ts
export interface Survey {
  id: string
  title: string
  description: string | null
  isPublished: boolean
  shareableCode: string
  createdAt: string
}

export interface CreateSurveyRequest {
  title: string
  description: string | null
}

export interface UpdateSurveyRequest {
  title: string
  description: string | null
  isPublished: boolean
}
